### PR TITLE
Don't throw on getMerklePath() failure

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Commit message explains it.

### CODE REVIEW

**Review by Geod24 — APPROVED** [review]

File: source/agora/node/Ledger.d [hunk 1]
```diff
@@ -419,13 +419,18 @@ public class Ledger
...
+    public Hash[] getMerklePath (ulong block_height, Hash hash) @safe nothrow
     {
         if (this.getBlockHeight() < block_height)
             return null;
 
         Block block;
-        this.storage.readBlock(block, block_height);
+        if (!this.storage.tryReadBlock(block, block_height))
+        {
+            log.trace("Could not read block at height {}: {}");
```
Comment (TrustHenry) [inline source/agora/node/Ledger.d]: `log.trace("Could not read block at height {}: {}", block_height, block );`
Is it okay if It don't have it?

Comment (Geod24) [reply source/agora/node/Ledger.d]: Oops

Comment (AndrejMitrovic) [reply source/agora/node/Ledger.d]: Good eyes @TrustHenry.

**Review by Geod24 — APPROVED** [review]

**Review by TrustHenry — APPROVED** [review]
LGTM

**Comment by AndrejMitrovic** [comment]
```
Running ./build/agora-unittests 
0   agora-unittests                     0x0000000102a374fe _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff7224b5fd _sigtramp + 29
0   agora-unittests                     0x0000000102a374fe _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff7224b5fd _sigtramp + 29
```

This segfault keeps appearing in many PRs. I think it's related to tasks, but I haven't figured out the root cause. And also, it's very hard to trigger it locally, making debugging a game of luck..

**Review by Geod24 — APPROVED** [review]

**Pull request merged**